engine/render/renderer/Core.h

```diff
@@ -36,24 +37,30 @@ typedef size_t size;
     std::cout << "                        Message: " << message << std::endl;
 
 // Custom assert macro
-#define CC_ASSERT(expr, message)                                   \
-    if (expr)                                                      \
-    {}                                                             \
-    else                                                           \
-    {                                                              \
-        REPORT_ASSERT_FAILURE(#expr, __FILE__, __LINE__, message); \
-        EXIT_APP                                                   \
-    }
+#define CC_ASSERT(expr, message) Siege::Assert(expr, __FILE__, __LINE__, message)
 
 #define OUT
 
 #define INTERN_STR(str) Utils::WSID(str)
 
 namespace Siege
 {
+inline void Assert(bool expr, const char* file, int line, const char* message)
+{
+    if (expr)
+    {}
+    else
+    {
+        REPORT_ASSERT_FAILURE(expr, file, line, message);
+        EXIT_APP
+    }
+}
+
```

Comment:
I'd just skip the middleman and not rely on the compiler to maybe inline if I ask nicely with:
```
#define CC_ASSERT(expr, message) Siege::Assert(expr, __FILE__, __LINE__, message)
     if (!expr) \
     { \
         REPORT_ASSERT_FAILURE(expr, file, line, message); \
         EXIT_APP \
     }
```

Reply:
I don't believe this will make a difference. The program will interpret this as a full statement and will put the `;` at the end of the `}` after the if statement. We'll still have the same issue with the semicolon not being needed.

Reply:
I'll add that if we do it the way you suggested then we'll end up with a bunch of warnings in Clion about empty statements.